python/lsst/drp/tasks/gbdesAstrometricFit.py

```diff
@@ -948,6 +962,47 @@ class `wcsfit.FoFClass`, associating them into matches as you go.
 
         return sourceIndices, columns
 
+    def _check_degeneracies(self, associations, extensionInfo):
+        """Check that the minimum number of visits and sources needed to
+        constrain the model are present.
+
+        This does not guarantee that the Hessian matrix of the chi-square,
+        which is used to fit the model, will be positive-definite, but if the
+        checks here do not pass, the matrix is certain to not be
+        positive-definite and the model cannot be fit.
+
+        Parameters
+        ----------
+        associations : `wcsfit.FoFClass`
+            Object holding the source association information.
+        extensionInfo : `lsst.pipe.base.Struct`
+            Struct containing properties for each extension.
+        """
+        # As a baseline, need to have more stars per detector than per-detector
+        # parameters, and more stars per visit than per-visit parameters.
+        whichExtension = np.array(associations.extn)
+        whichDetector = np.zeros(len(whichExtension))
+        whichVisit = np.zeros(len(whichExtension))
+
+        for extension, (detector, visit) in enumerate(zip(extensionInfo.detector, extensionInfo.visit)):
+            ex_ind = whichExtension == extension
+            whichDetector[ex_ind] = detector
+            whichVisit[ex_ind] = visit
+
+        if "BAND/DEVICE/poly" in self.config.deviceModel:
+            nCoeffDetectorModel = int(((2 * self.config.devicePolyOrder + 3) ** 2 - 1) / 8)
+            unconstrainedDetectors = []
+            for detector in np.unique(extensionInfo.detector):
+                numSources = (whichDetector == detector).sum()
+                if numSources < nCoeffDetectorModel:
+                    unconstrainedDetectors.append(str(detector))
+
+            if unconstrainedDetectors:
+                raise RuntimeError(
+                    "The model is not constrained. The following detectors do not have enough "
+                    f"sources ({nCoeffDetectorModel} required): " + ", ".join(unconstrainedDetectors)
```

Comment:
Because this is within parenthesis, you could drop the `+` and the strings would concatenate anyway.